moveit_planners/stomp/include/stomp_moveit/math/multivariate_gaussian.hpp

```diff
@@ -79,9 +75,8 @@ class MultivariateGaussian
   Eigen::MatrixXd covariance_cholesky_; /**< Cholesky decomposition (LL^T) of the covariance */
 
   int size_;
-  boost::mt19937 rng_;
-  boost::normal_distribution<> normal_dist_;
-  std::shared_ptr<boost::variate_generator<boost::mt19937, boost::normal_distribution<> > > gaussian_;
+  std::mt19937 rng_;
```

Comment:
I think the rng's seed needs to be randomized. E.g. with random_device:
```
    std::random_device rd{};
    std::mt19937 rng{rd()};
```
If you don't explicitly choose, the random sequence will be always the same since it is initialized with 5489u ([reference](https://en.cppreference.com/w/cpp/numeric/random/mersenne_twister_engine)). Try running this example in [godbolt](https://godbolt.org/) and then replace `std::mt19937 rng` with the two lines above

```cpp
// Type your code here, or load an example.
#include <random>
#include <iostream>

int main(int argc, char** argv)  {
    std::mt19937 rng;
    std::normal_distribution<double> normal_dist;
    
    for(auto i = 0; i < 10; i++){
        std::cout << normal_dist(rng) << std::endl;
    }
    return 0;
}
```
You can re-run the program with the small :arrow_right: icon in the bottom left

Reply:
You're right @sjahr , I'll make the changes in the evening.